### PR TITLE
Polyfill a more friendly i18n API for device

### PR DESCRIPTION
**Pull request by corhere**
There's a lot of refactoring required to make this work while keeping the interfaces clean. I recommend reviewing each commit individually.

### CODE REVIEW

File: src/__test__/compile/importPackage.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import jszip from 'jszip';
```
Comment (corhere) [inline src/__test__/compile/importPackage.js]: JSZip is big, and takes nearly a second to compile on my machine. Switching to a smaller package for this test case shaves that much off each test run.

File: src/compile.ts [hunk 2]
```diff
@@ -34,130 +33,81 @@ const tsconfigOverrides = {
...
+        plugins: [
+          typescript({
+            onDiagnostic,
+            tsconfigOverride: {
+              ...tsconfigOverrides,
+              target: ecma === 6 ? ts.ScriptTarget.ES2015 : ts.ScriptTarget.ES5,
+            },
+          }),
+          ...pluginIf(component === ComponentType.DEVICE, polyfillDevice),
+          ...pluginIf(component === ComponentType.SETTINGS, resourceImports),
```
[the remark this answers is not in the document]
Comment (corhere) [reply src/compile.ts]: Breaking change or bugfix depending on your perspective. Should I gate it on SDK version?

File: src/plugins/polyfillDevice.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import polyfill from './polyfill';
+
+const modules = {
+  i18n: `
+import { getResource } from 'resources';
+
+export function gettext(msgid) {
```
[the remark this answers is not in the document]
Comment (corhere) [reply src/plugins/polyfillDevice.ts]: I'm with you there, though loading the polyfill modules off disk does cause some headaches for the Studio build of toolchain. I wanted to delay having to deal with that for now.

[the remark this answers is not in the document]
Comment (Hexxeh) [reply src/plugins/polyfillDevice.ts]: Yeah I forgot about the Studio case here, makes sense.

File: src/compile.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hexxeh) [reply src/compile.ts]: Given it's pretty trivial for us to do so, I think we should.

**Pull request merged**